### DISCUSSION TITLE
Group and Permission based on database

### DISCUSSION THREAD

**Discussion by tomchandra** [codeigniter4/shield#562] [Q&A] [answered]
any chance to make master group and permission save to database and link that to table auth_group_users/auth_permissions_user like myth auth method, so we dont need defining availible permission or group on shield config. thanks

**Comment by lonnieezell** [comment, marked as answer]
You can use the settings() helper to save it to the database and it will work without changes. 

The system was designed to be able to commit the groups and permissions to version control for easier deployments but we built the settings library to allow that to be overridden at runtime.

[Bonfire](https://github.com/lonnieezell/Bonfire2) has an example of this in its group settings if you need a reference.

**Reply by bgeneto** [reply]
Sorry, I believe this is not the proper answer. Please correct me if I'm wrong... AFAIK bonfire2 has no CRUD for users' groups and this is not a trivial task to accomplish as it seems. 
Since one cannot delete a single key from the `AuthGroups.groups` array in settings with `forget()`, an update operation that changes e.g. a name/key have to delete the whole `AuthGroups.group` array and insert it again 'guessing' which key have changed. 
I really miss a real relational database table for auth groups and permissions in Shield.
